pytorch-example/run_pytorch.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Function to detect Linux distribution
+detect_distro() {
+    if [ -f /etc/os-release ]; then
+        . /etc/os-release
+        echo $ID
+    elif [ -f /etc/redhat-release ]; then
+        echo "rhel"
+    elif [ -f /etc/debian_version ]; then
+        echo "debian"
+    else
+        echo "unknown"
+    fi
+}
+
+# Install system dependencies based on distribution
+DISTRO=$(detect_distro)
+
+case $DISTRO in
+    "fedora"|"rhel"|"centos"|"rocky"|"almalinux")
+        if command -v dnf >/dev/null 2>&1; then
+            dnf install gcc-toolset-13 python3.11-devel python3.11-pip -y --skip-broken --nobest
+            source /opt/rh/gcc-toolset-13/enable
+        else
+            yum install gcc-toolset-13 python3.11-devel python3.11-pip -y
+            source /opt/rh/gcc-toolset-13/enable
+        fi
+        ;;
+    "ubuntu"|"debian")
+        # Use: bash script.sh
+        apt update -y
+        apt install -y software-properties-common curl lsb-release gnupg2  libgomp1
+
+        add-apt-repository ppa:deadsnakes/ppa -y
+        apt update -y
+        apt install -y python3.11 python3.11-venv python3.11-distutils libgfortran5 gcc-13
+        ;;
```

Comment:
This code assumes it'd be run as `root`, and I'm not sure if that's a good assumption. Why don't we use `sudo` for running system package install commands?